### PR TITLE
Adding hacluster support

### PR DESCRIPTION
**Pull request by hyperbolic2346**
**What type of PR is this?**
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:
Adds hacluster support
**Special notes for your reviewer**:
Requires https://github.com/juju-solutions/layer-hacluster/pull/1, https://github.com/juju-solutions/interface-public-address/pull/3, and https://github.com/juju-solutions/interface-http/pull/17

```release-note
Added support for hacluster, which allows for virtual IPs in front of the control plane with or without load balancers.
```

### CODE REVIEW

File: cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py [hunk 1]
```diff
@@ -948,11 +992,19 @@ def addons_ready():
 
 @when('loadbalancer.available', 'certificates.ca.available',
       'certificates.client.cert.available', 'authentication.setup')
-def loadbalancer_kubeconfig(loadbalancer, ca, client):
+def loadbalancer_kubeconfig():
```
Comment (Cynerva) [inline cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: This is called here, which needs to be updated:
https://github.com/juju-solutions/kubernetes/blob/a15b713653127a5fa515ccfa4bbc03561eccea2d/cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py#L2171

File: cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py [hunk 2]
```diff
@@ -962,19 +1014,25 @@ def loadbalancer_kubeconfig(loadbalancer, ca, client):
 @when('certificates.ca.available', 'certificates.client.cert.available',
       'authentication.setup')
 @when_not('loadbalancer.available')
-def create_self_config(ca, client):
+def create_self_config():
```
Comment (Cynerva) [inline cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: This is called here, which needs to be updated: https://github.com/juju-solutions/kubernetes/blob/a15b713653127a5fa515ccfa4bbc03561eccea2d/cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py#L2173

**Comment by Cynerva** [comment]
You might want to change the selection behavior for kube-proxy, too:

https://github.com/juju-solutions/kubernetes/blob/a15b713653127a5fa515ccfa4bbc03561eccea2d/cluster/juju/layers/kubernetes-common/lib/charms/layer/kubernetes_common.py#L484

You might even be able to just remove that line, since kube-proxy is configured with --kubeconfig and will presumably use the server in there if --master is not specified.

**Comment by hyperbolic2346** [comment]
Needs to wait for https://review.openstack.org/#/c/637205/

**Comment by Cynerva** [comment]
Please open a PR to add layer-hacluster to https://github.com/juju/layer-index. We'll need it in there before we can merge this.

**Comment by Cynerva** [comment]
Once layer-hacluster is added to the layer index, this LGTM

**Pull request merged**